### ISSUE TITLE
Data export broken

### ISSUE THREAD

**Issue by andersfi** [gbif-norway/dugnad#12]
"cannot open URL 'https://dugnad.gbif.no/nb_NO/project/huitfeldt-kaas/export.json': HTTP status was '500 Internal Server Error'"

**Comment by umeldt** [comment]
Looks like I no longer have access to the gbif.no VM and there's not much I can do without more information...

Maybe @dagendresen or @rukayaj can check the web server's log files? There should be a more informative error message somewhere.

**Comment by rukayaj** [comment]
I had a look and I don't see anything for dugnad in the lighttpd error.log, only a few messages to do with rdflib and the resolver. The access log shows that dugnad is getting accessed and returning 500 for that URL, but nothing else.

I notice that any chars appended to the URL https://dugnad.gbif.no/nb_NO/project/huitfeldt-kaas/ result in a 500. I also don't see any mention of "export" in the source code. Is this definitely the right URL to get an export? When were you last able to get a json export @andersfi, is it something that broke recently?

**Comment by andersfi** [comment]
The last successful download I can find a record of is 7th of May 2019 20:40. The URL used for that was https://dugnad.gbif.no/nb_NO/project/huitfeldt-kaas/export.json

**Comment by umeldt** [comment]
Thanks for taking a look, @rukayaj!

There are two instances of dugnad running - https://data.gbif.no/dugnad/ (which is also giving me a 500 now - it didn't a couple of weeks ago) and https://dugnad.gbif.no/. Only the second instance has JSON export code.
I don't remember exactly where the source code for each instance is located... should be somewhere under /gbif, though.

I'm guessing this is either some kind of unicode/charset bug in the dugnad code -or- an SELinux issue. Security updates from USIT would sometimes reset SELinux security contexts on the VM...

**Comment by rukayaj** [comment]
Ok yes, I was looking at the source code in /gbif/dugnad, and it looks like I should have been looking in /gbif/neodugnad. 

I thought it might be an SELinux thing because there's this in the audit logs
`type=AVC msg=audit(1537054631.485:763177): avc:  denied  { write } for  pid=32694 comm="dugnad.py" name="projects.db" dev="sdb1"`. But when I followed the MUSIT instructions to make the relevant policy changes it didn't seem to make any difference and the 500 error is still there. And I guess just outputting json wouldn't mean the app would be trying to write to projects.db. So maybe I have fixed something there, but not this issue.

Then I realised there's another project - https://dugnad.gbif.no/nb_NO/project/gunnerus/export.json which is working, so it probably is a funny character in the huitfeldt-kaas output somewhere. To debug this a bit more it would be nice to have working logs, so I looked into that too. It seems lighttpd is not serving the app, just the static files: in lighttpd conf 

```
$HTTP["host"] == "dugnad.gbif.no" { 
  server.document-root = "/gbif/www/dugnad" 
  $HTTP["url"] !~ "^/(static|.well-known)/" {
     proxy.server = ( "" => (( "host" => "127.0.0.1", "port" => 8081 )))
   }
...
```

So i guess everything that's not static or .well-known is using gunicorn, which explains why there's no logs in lighttpd. I think I can turn on logging in the bottle framework by passing in debug=True to the run method to get logs, but I need to look at something else now so I can only try tomorrow. 

The other thing that's a bit weird is that there are two dugnad processes running (three including lighttpd one):
```
[rukayasj@data neodugnad]$ sudo ps aux | grep dugnad
lighttpd  7516  0.0  0.5 713876 21596 ?        Sl   13:06   0:05 /usr/bin/python /gbif/dugnad/dugnad.py
root     13589  0.0  1.7 1296164 67024 pts/6   S+   May28   7:12 /usr/bin/python ./dugnad.py --bind 127.0.0.1:8081 -t 240
root     29794  0.0  0.0 254968  1196 pts/6    S+   Jan31  35:34 /usr/bin/python ./dugnad.py --bind 127.0.0.1:8081 -t 240
```

So maybe the one started May28 is unnecessary and is somehow causing problems. Unless there should be two running for some reason @umeldt ?
